internal/onprem/hydrate.go

```diff
@@ -160,23 +160,25 @@ func (h *OnPremEventsHydrator) downloadURL(url string, msg kafka.Message) {
 
 func (h *OnPremEventsHydrator) ProcessMessage(ctx context.Context, msg *kafka.Message) error {
 	for _, header := range msg.Headers {
-		if header.Key == "service" {
-			service := string(header.Value)
-			if service == "assisted-installer" {
-				payload := OnPremPayload{}
-				err := json.Unmarshal(msg.Value, &payload)
-				if err != nil {
-					h.logger.WithFields(logrus.Fields{
-						"msg": msg,
-					}).Warning("could not decode message value")
-					continue
-				}
+		if header.Key != "service" {
+			h.ackChannel <- *msg
+			continue
+		}
+		service := string(header.Value)
+		if service == "assisted-installer" {
```

Comment:
Sorry, just saw this! How about messages that don't have `assisted-installer` as the header Value? Do we ack those in another place or not have to worry about them?

Reply:
Good question, messages with `assisted-installer` were already acked [here](https://github.com/openshift-assisted/assisted-events-stream/blob/master/internal/onprem/hydrate.go#L132) after processing, but I forgot to ack non `assisted-installer` messages.
The problem with this, is that if the consumer restarts, will read again all messages not related to assisted-installer (functionally OK, but inefficient)